flocker/control/test/test_httpapi.py

```diff
@@ -437,6 +437,68 @@ def created(ignored):
         return creating
 
 
+class UpdateDatasetTestsMixin(APITestsMixin):
+    """
+    Tests for the dataset modification endpoint at ``/datasets/<dataset_id>``.
+    """
+    def test_unknown_dataset(self):
+        """
+        NOT_FOUND is returned if the requested dataset_id doesn't exist.
+        The error includes the requested dataset_id.
+        """
+
+    def test_primary_changed(self):
+        """
+        If a different primary IP address is supplied, the modification request
+        succeeds.
+        """
+        # Pre-populate the persistence service with a primary other_manifestation on node1
+        # Request it move to node2.
+        # Assert that the returned code is OK
+
+    def test_primary_unchanged(self):
+        """
+        If the current primary IP address is supplied, the modification request
+        succeeds.
+        """
+        # Pre-populate the persistence service with a primary other_manifestation on node1
+        # Request it move to node1.
+        # Assert that the returned code is OK
+
+    def test_primary_unknown(self):
+        """
+        A dataset's primary IP address must belong to a node in the cluster.
+        XXX: Skip this test until FLOC-1278 is implemented.
+        """
+
+    def test_primary_invalid(self):
```

Comment:
Well, it's useful to have at least one test that ensures JSON schema is actually being checked... OTOH this isn't it, since hostname is going to go away at some point.